hcl/hcl.go

```diff
@@ -1002,6 +1023,115 @@ func parseStack(evalctx *eval.Context, stack *Stack, stackblock *ast.Block) erro
 	return errs.AsError()
 }
 
+func checkHasSubBlocks(block *ast.Block, blocks ...string) error {
+	errs := errors.L()
+
+	for _, got := range block.Attributes.SortedList() {
+		errs.Append(errors.E(got.NameRange,
+			"unrecognized attribute %s.%s", block.Type, got.Name,
+		))
+	}
+
+	for i, label := range block.Labels {
+		errs.Append(errors.E(ErrTerramateSchema,
+			"block %s has unexpected label %s",
+			block.Type,
+			block.LabelRanges[i],
+			label))
+	}
+
+	found := false
+checkBlocks:
+	for _, got := range block.Blocks {
+		for _, want := range blocks {
+			if want == got.Type {
+				if found {
+					errs.Append(errors.E(ErrTerramateSchema,
+						got.DefRange(),
+						"duplicated block %s",
+						got.Type),
+					)
+					continue checkBlocks
```

Comment:
```suggestion
```
This line seems unneeded.

Reply:
I wanted to avoid setting found again to true. it doesn't change the semantics (it is already true), but it seemed more well behaved, maybe in the future something else is done after this check that doesn't make sense when it is a duplicate.